frontend/src/features/apartment/ApartmentDetailsPage.tsx

```diff
@@ -70,26 +73,56 @@ const ApartmentDetailsPage = (): JSX.Element => {
                         {data.address.stair}
                         {data.address.apartment_number}
                     </span>
-                    <span>{data.type.value}</span>
+                    <span>
+                        2 {/* TODO: Substitute with dynamic value once the API supplies one */}
+                        {data.type.value}
+                    </span>
                     <span>{data.surface_area}m²</span>
                     <span>{data.address.floor}.krs</span>
                 </h2>
                 <div className="apartment-action-cards">
                     <Card>
-                        <p>Vahvistamaton enimmäishinta</p>
-                        <p className="price">210 000 €</p>
-                        <p>Vahvistettu enimmäishinta</p>
-                        <p className="price">-</p>
+                        <label className="card-heading">Vahvistamaton enimmäishinta</label>
+                        <div className="unconfirmed-prices">
+                            <div className="price">
+                                <span className="basis">Markkinahintaindeksi</span>
+                                <span className="amount">
+                                    <span className="value">{priceMH}</span> €
+                                </span>
+                            </div>
+                            <div className="price price--current-top">
+                                <span className="basis">Rakennushintaindeksi</span>
+                                <span className="amount">
+                                    <span className="value">{priceRH}</span> €
+                                </span>
+                            </div>
+                            <div className="price">
+                                <span className="basis">Rajaneliöhinta</span>
+                                <span className="amount">
+                                    <span className="value">{priceRNH}</span> €
+                                </span>
+                            </div>
+                        </div>
+                        <label className="card-heading">Vahvistettu enimmäishinta</label>
+                        <p className="confirmed-price">250000€</p>
+                        <Button
+                            className="button-confirm"
+                            theme="black"
+                            size="small"
+                            style={{display: "none"}} // Hidden until it's
```

Comment:
..haha, indeed! The missing word is: needed :D 
Thought it might be good to not have the button there to confuse the client, while its operation is still a bit of a mystery (..and as it does nothing). Just wanted to have a readiness for the button in question.